sdk/cosmos/azure-cosmos/azure/cosmos/_retry_utility.py

```diff
@@ -112,6 +129,33 @@ def Execute(client, global_endpoint_manager, function, *args, **kwargs):
                 retry_policy = sessionRetry_policy
             elif exceptions._partition_range_is_gone(e):
                 retry_policy = partition_key_range_gone_retry_policy
+            elif exceptions._container_recreate_exception(e):
+                retry_policy = container_recreate_retry_policy
+                # pylint: disable=protected-access
+                # Before we retry if retry policy is container recreate, we need refresh the cache of the
+                # container properties and pass in the new RID in the headers.
+                client._refresh_container_properties_cache(retry_policy.container_link)
+                if retry_policy.check_if_rid_different(
```

Comment:
I am curious why we didn't use `if` condition based on sub-status codes - it enhances readability or knowing what business logic to apply for a given error code combination.

Reply:
The sub status that are not RID_MISMATCH can be a result of a regular error not related to container recreate, and in those cases we want to avoid to retry as they would usually not be a retryable error. (They're both 404s), checking if the rid is different (usually what RID_MISMATCH would tell us) would tell us with certainty that it was due to a container recreate which means we should retry. It would make sense to check if the sub status code is RID_MISMATCH as we can skip checking the RIDs in that case.